src/System/RosImplementation/src/RosClock.cpp

```diff
@@ -0,0 +1,70 @@
+/**
+ * @file RosClock.cpp
+ * @authors Giulio Romualdi
+ * @copyright 2022 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the BSD-3-Clause license.
+ */
+
+#include <thread>
+
+#include <rclcpp/clock.hpp>
+#include <rclcpp/rclcpp.hpp>
+
+#include <BipedalLocomotion/System/RosClock.h>
+
+using namespace BipedalLocomotion::System;
+
+std::chrono::duration<double> RosClock::now()
+{
+    return std::chrono::duration<double>(m_clock.now().seconds());
+}
+
+void RosClock::sleepFor(const std::chrono::duration<double>& sleepDuration)
+{
+    // std::chrono::duration store the time in second
+    m_clock.sleep_for(sleepDuration);
+}
+
+void RosClock::sleepUntil(const std::chrono::duration<double>& sleepTime)
+{
+    m_clock.sleep_for(sleepTime - this->now());
+}
+
+void RosClock::yield()
+{
+    std::this_thread::yield();
```

Comment:
Does it make sense to mix ROS2 and ``std`` calls? Is it better to simply do anything?

Reply:
You're right. Removed

Reply:
Not a big problem, but it could make sense to have `std::this_thread::yield()` in this context. In the end, `sleep`/`sleepUntil` are functions that in the end are waiting on some kind of OS synchronization primitive (mutex/condition variables), so it could make sense that you want to yield even if you are waiting on those. Anyhow, I do not think it is a big problem.

Reply:
I just thought that the effect could have not been very predictable given the sleep implementation of ROS2, but maybe I am being paranoid.

Reply:
In ros we should use the command `spin` so in theory yield is not needed. Shall we add a debug print that remember the user that this implementation of yield does nothing?

Reply:
> Shall we add a debug print that remember the user that this implementation of yield does nothing?

I would avoid the print, as it would make writing generic code quite difficult (you would need to not call yield if you use the ROS2 clock). I would say let's merge it empty (safest choice), and then let's see if problems emerge.

Reply:
Ok merged thank you all as usual ❤️